bn-apps/cordapp-updates-distribution/corda-updates-transport/src/main/kotlin/net/corda/cordaupdates/transport/flows/PeekResourceFlow.kt

```diff
@@ -0,0 +1,63 @@
+package net.corda.cordaupdates.transport.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import net.corda.cordaupdates.transport.flows.Utils.toCordaException
+import net.corda.cordaupdates.transport.flows.Utils.transporterFactory
+import net.corda.core.flows.FlowLogic
+import net.corda.core.flows.FlowSession
+import net.corda.core.flows.InitiatedBy
+import net.corda.core.flows.InitiatingFlow
+import net.corda.core.flows.StartableByRPC
+import net.corda.core.flows.StartableByService
+import net.corda.core.identity.CordaX500Name
+import net.corda.core.utilities.unwrap
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils
+import org.eclipse.aether.repository.RemoteRepository
+import org.eclipse.aether.spi.connector.transport.PeekTask
+import java.lang.Exception
+import java.net.URI
+
+/**
+ * Peeks resource in a remote repository. Used internally by Maven Resolver.
+ *
+ * @param resourceLocation location of the resource provided by Maven Resolver
+ * @param repoHosterName x500Name of the repository hoster
+ */
+@InitiatingFlow
+@StartableByRPC
+@StartableByService
+class PeekResourceFlow(private val resourceLocation : String, private val repoHosterName : String) : FlowLogic<Unit>() {
+    @Suspendable
+    override fun call() {
+        val repoHosterParty = serviceHub.identityService.wellKnownPartyFromX500Name(CordaX500Name.parse(repoHosterName))!!
+        val repoHosterSession = initiateFlow(repoHosterParty)
+        repoHosterSession.sendAndReceive<Boolean>(resourceLocation).unwrap { it }
```

Comment:
What's the boolean for? Why is it being ignored?

Reply:
If the resource is missing - then a `ResourceNotFoundException` would be thrown. If the resource exists - then we are not interested in what the actual reply is. Cha

Reply:
Can you put the explanation in the code please.

Reply:
Done